dgl_sparse/src/matrix_ops.cc

```diff
@@ -58,12 +55,106 @@ std::tuple<std::shared_ptr<COO>, torch::Tensor, torch::Tensor> COOIntersection(
   return {ret_coo, lhs_indices, rhs_indices};
 }
 
-std::tuple<c10::intrusive_ptr<SparseMatrix>, torch::Tensor> Compact(
-    const c10::intrusive_ptr<SparseMatrix>& mat, uint64_t dim,
-    torch::Tensor leading_indices) {
-  DGL_SPARSE_COO_SWITCH(mat->COOPtr(), XPU, IdType, "Compact", {
-    return CompactImpl<XPU, IdType>(mat, dim, leading_indices);
-  });
+std::tuple<torch::Tensor, torch::Tensor> CompactIndices(
+    const torch::Tensor& row,
+    const torch::optional<torch::Tensor>& leading_indices) {
+  torch::Tensor sort_row, sort_idx;
+  std::tie(sort_row, sort_idx) = row.sort(-1);
+  torch::Tensor rev_sort_idx = torch::empty_like(sort_idx);
+  rev_sort_idx.index_put_({sort_idx}, torch::arange(0, sort_idx.numel()));
```

Comment:
```suggestion
  rev_sort_idx.index_put_({sort_idx}, torch::arange(0, sort_idx.numel(), device=sort_idx.device));
```

Reply:
It doesn't seem to have arguments like 'device'.